.circleci/aggregate-build-artifacts.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -e 
+
+modules_with_build_trees() {
+   for item in *; do
+       if [ -d "$item/build" ]; then
+           echo "$item"
+       fi
+   done
+}
+readonly modules=($(modules_with_build_trees))
+readonly current_dir="$(pwd)"
+readonly aggregation_dir="build-artifacts"
+
+if [ ! -d "$aggregation_dir" ]; then
+    mkdir "$aggregation_dir"
```

Comment:
`mkdir -p` would be slightly more robust in that it a) will make the full tree if the spec happens to need it, and b) won't fail if the directory suddenly starts to exist between the test and the `mkdir` command.

Reply:
will move to this